### PR TITLE
test(go): disable caching Job objects

### PR DESCRIPTION
**Pull request by padlar**
Signed-off-by: Sunil Shivanand <sunil.shivanand@statnett.no>

### CODE REVIEW

File: controllers/suite_test.go [hunk 1]
```diff
@@ -178,3 +178,16 @@ func newPod(owner client.Object, scheme *runtime.Scheme) *corev1.Pod {
...
 }
+
+func newNoCacheClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
+	c, err := client.New(config, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.NewDelegatingClient(client.NewDelegatingClientInput{
+		CacheReader:     nil,
```
Comment (erikgb) [inline controllers/suite_test.go:189]:
```suggestion
```

Isn't this the zero-value?

File: controllers/suite_test.go [hunk 2]
```diff
@@ -178,3 +178,16 @@ func newPod(owner client.Object, scheme *runtime.Scheme) *corev1.Pod {
...
+		Client:          c,
+		UncachedObjects: uncachedObjects,
```
Comment (erikgb) [inline controllers/suite_test.go:191]:
```suggestion
```

No need to forward this, as the cache is now fully disabled? Or?

**Review by erikgb — CHANGES_REQUESTED** [review]

File: controllers/suite_test.go (hunk 1 above)
Comment (padlar) [reply controllers/suite_test.go]: updated to just disable caching Job objects

File: controllers/suite_test.go [hunk 3]
```diff
@@ -89,8 +89,9 @@ var _ = BeforeSuite(func() {
 		Expect(err).To(Succeed())
 	}
 
+	uncachedObjects := []client.Object{&batchv1.Job{}}
```
Comment (erikgb) [inline controllers/suite_test.go:92]: Maybe add a comment why we are doing this? If this fixes the issue, we should/must follow this up further...

**Review by erikgb — CHANGES_REQUESTED** [review]

File: controllers/suite_test.go (hunk 3 above)
Comment (padlar) [reply controllers/suite_test.go:93]: added a fixme

**Review by erikgb — APPROVED** [review]
LGTM

**Pull request merged**